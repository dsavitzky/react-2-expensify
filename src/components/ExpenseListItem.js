import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem =  ({ description, amount, createdAt, id }) => (
    <li>
        <p>Description: <Link to={`/edit/${id}`}>{description}</Link></p>
        <p>Amount: {numeral(amount / 100).format('$0,0.00')}</p>
        <p>Created at: {moment(createdAt).format('MMMM Do, YYYY')}</p>
    </li>
)

export default ExpenseListItem