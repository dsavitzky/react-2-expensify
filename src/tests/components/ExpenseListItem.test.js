import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import { expenses } from '../fixtures/expenses'
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

test('should render ExpenseListItem with expense', () => {

    const wrapper = shallow(<ExpenseListItem {...expenses[1]} />)
    expect(wrapper).toMatchSnapshot()
})