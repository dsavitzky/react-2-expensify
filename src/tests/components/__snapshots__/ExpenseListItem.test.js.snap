// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expense 1`] = `
<li>
  <p>
    Description: 
    <Link
      replace={false}
      to="/edit/2"
    >
      Rent
    </Link>
  </p>
  <p>
    Amount: 
    $5.00
  </p>
  <p>
    Created at: 
    January 5th, 1970
  </p>
</li>
`;
